repos/keg-components/src/components/form/checkGroup/checkGroup.js

```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import { View } from 'KegView'
+import { Text } from '../../typography/text'
+import { Checkbox } from 'KegCheckbox'
+import { useStyle } from '@keg-hub/re-theme'
+import { useClassList } from 'KegClassList'
+import { useChildrenWithRefs } from '../../../hooks/useChildrenWithRefs'
+import { mapObj } from '@keg-hub/jsutils'
+
+/**
+ * Simple header for CheckGroup, without a checkbox
+ * @param {Object} props
+ * @param {string} props.title - title of header
+ * @param {Object} props.style - style rules to apply to the Text element
+ * @param {*} props.* - remaining props are passed to the Text element
+ */
+const SimpleHeader = ({ title, className, style, ...rest }) => {
+  const textStyle = useStyle('form.checkGroup.simpleHeader.main', style)
+  return (
+    <Text
+      className={className}
+      style={textStyle}
+      {...rest}
+    >
+      { title }
+    </Text>
+  )
+}
+
+/**
+ * Header for CheckGroup, with a checkbox for select-all functionality
+ * @param {Object} props
+ * @param {string} props.title - title of header
+ * @param {Object} props.style - style rules to apply to the header
+ * @param {Function?} props.onPress - handler for checkbox press
+ * @param {boolean} props.checked - initial value of header checkbox
+ */
+const CheckboxHeader = ({ title, className, style, onPress, checked }) => {
+  const headerStyles = useMemo(
+    () => ({
+      main: style,
+      content: {
+        right: {},
+      },
+    }),
+    [style]
+  )
+
+  const onChangeHandler = useCallback((_, val) => onPress?.(val), [onPress])
+
+  return (
+    <Checkbox
+      RightComponent={title}
+      rightClassName={className}
+      styles={headerStyles}
+      checked={checked}
+      onChange={onChangeHandler}
+      close
+    />
+  )
+}
+
+/**
+ * A group of checkbox items with a header.
+ * Will include a select-all checkbox in the header IF you define the children as a function (@see story examples)
+ * @param {Object} props
+ * @param {string} props.className - css class name
+ * @param {string} props.headerClassName - css class name for header text
+ * @param {string} props.title - title of the group
+ * @param {boolean?} props.initChecked - initial checked value of the group header box, if you are using it
+ * @param {boolean?} props.check - true if you want a group header check box
```

Comment:
This prop name seems odd `check` to me does not indicate it shows the GroupHeader checkBox
Also I would add a way to disable the header completely.
There may be cases where I don't want it, but I want the checkbox group

Reply:
updated 
- changed prop name to `showHeaderCheckbox`
- added prop `showHeader`, true by default, but hides the header completely when false